src/gpu/sycl/ref_reorder.hpp

```diff
@@ -0,0 +1,115 @@
+/*******************************************************************************
+* Copyright 2022-2024 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef GPU_SYCL_REF_REORDER_HPP
+#define GPU_SYCL_REF_REORDER_HPP
+
+#include "gpu/gpu_reorder_pd.hpp"
+#include "gpu/sycl/sycl_gpu_primitive.hpp"
+#include "gpu/sycl/sycl_io_helper.hpp"
+#include "gpu/sycl/sycl_post_ops.hpp"
+#include "gpu/sycl/sycl_primitive_conf.hpp"
+#include "gpu/sycl/sycl_q10n.hpp"
+#include "sycl/sycl_stream.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace gpu {
+namespace sycl {
+
+struct ref_reorder_t : public sycl_gpu_primitive_t {
+    using sycl_gpu_primitive_t::sycl_gpu_primitive_t;
+
+    struct pd_t : public gpu_reorder_pd_t {
+        using gpu_reorder_pd_t::gpu_reorder_pd_t;
+
+        DECLARE_COMMON_PD_T("dpcpp:ref:any", ref_reorder_t);
+
+        status_t init(
+                engine_t *engine, engine_t *src_engine, engine_t *dst_engine) {
+            using namespace data_type;
+            using sm = primitive_attr_t::skip_mask_t;
+
+            const memory_desc_wrapper src_d(src_md(0));
+            const memory_desc_wrapper dst_d(dst_md());
+
+            const bool ok = check_data_types(src_d, dst_d)
+                    && check_formats(src_d, dst_d)
+                    && attr()->has_default_values(
+                            sm::scales_runtime | sm::post_ops)
+                    && (src_md(0)->format_desc.blocking.inner_nblks == 0)
+                    && (dst_md()->format_desc.blocking.inner_nblks == 0)
```

Comment:
Minor: better be a part of `check_formats`.

Reply:
This was actually a duplicate of what is checked in `check_formats`. Removed.